lnd_test.go

```diff
@@ -13852,6 +13859,134 @@ func testChannelBackupRestore(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 }
 
+// testTLSAutoRegeneration creates an expired TLS certificate, to test that a 
+// new TLS certificate pair is regenerated when the old pair expires. This is
+// necessary because the pair expires after a little over a year.
+func testTLSAutoRegeneration(lnNet *lntest.NetworkHarness, t *harnessTest) {
+	certPath := lnNet.Alice.TLSCertStr()
+	keyPath := lnNet.Alice.TLSKeyStr()
+
+	// Create an expired certificate.
+	expiredCert := genExpiredCertPair(
+		t, lnNet, certPath, keyPath,
+        )
+
+	// Restart the node to test that the test is automatically regenerated.
+	lnNet.RestartNode(lnNet.Alice, nil, nil)
+
+	// Grab the newly generated certificate.
+	newCertData, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		t.Fatalf("couldn't grab new certificate")
+	}
+
+	newCert, err := x509.ParseCertificate(newCertData.Certificate[0])
+	if err != nil {
+		t.Fatalf("couldn't parse new certificate")
+	}
+
+	// Test that the expired certificate was deleted and a new certificate was generated.
+	if expiredCert == newCert {
```

Comment:
is it possible for the two certs to share the same pointer? o/w i think just the last check is sufficient

Reply:
Good point... just tweaked those two things :)